lib/carrierwave/storage/postgresql_lo.rb

```diff
@@ -71,11 +71,12 @@ def store!(file)
 
       def retrieve!(identifier)
         raise "This uploader must be mounted in an ActiveRecord model to work" unless uploader.model
+        @oid = identifier
         CarrierWave::Storage::PostgresqlLo::File.new(uploader)
       end
 
       def identifier
-        @oid ||= uploader.model.read_attribute(uploader.mounted_as) || connection.lo_creat
+        @oid ||= connection.lo_creat
```

Comment:
I have not been able to create a failing test using the stubs that are currently used in the tests. I have created a full stack test at https://github.com/diogob/carrierwave-postgresql/compare/master...lsylvester:storage-should-not-depend-on-column-it-was-mounted-on-test which fails before and passes after. 

Is a full stack test ok with you?